hack/test

```diff
@@ -3,60 +3,32 @@
 set -eu -o pipefail
 
 : "${BUILDX_CMD=docker buildx}"
-: "${CACHE_FROM=}"
-: "${CACHE_TO=}"
-
-: ${TEST_INTEGRATION=}
-: ${TEST_REPORT_SUFFIX=}
-: ${TEST_KEEP_CACHE=}
-: ${TEST_DOCKERD=}
-: ${TEST_BUILDKIT_IMAGE=}
-
-if [ -n "$CACHE_FROM" ]; then
-  for cfrom in $CACHE_FROM; do
-    setFlags+=(--set "*.cache-from=$cfrom")
-  done
-fi
-if [ -n "$CACHE_TO" ]; then
-  for cto in $CACHE_TO; do
-    setFlags+=(--set "*.cache-to=$cto")
-  done
-fi
 
-if [ "$#" == 0 ]; then TEST_INTEGRATION=1; fi
+: "${TEST_IMAGE_BUILD=1}"
+: "${TEST_IMAGE_ID=buildx-tests}"
 
-while test $# -gt 0; do
-  case "$1" in
-    integration)
-      TEST_INTEGRATION=1
-      ;;
-    *)
-      echo "unknown arg $1"
-      ;;
-  esac
-  shift
-done
```

Comment:
I think I'd like to keep this if possible, we still default to integration if nothing else is specified.

I think we're going to need to create another suite of integration tests at some point to test the complexities of driver creation, e.g. there are a lot of options for setting up the kubernetes driver. These tests don't test the commands package directly, so it would make more sense to split it out, so we can run them in parallel etc.

Reply:
> I think I'd like to keep this if possible, we still default to integration if nothing else is specified.

On BuildKit we needed this special case because we got to test the frontend specifically and also the gateway and I don't think we need to have this kind of logic in buildx.

> I think we're going to need to create another suite of integration tests at some point to test the complexities of driver creation, e.g. there are a lot of options for setting up the kubernetes driver. These tests don't test the commands package directly, so it would make more sense to split it out, so we can run them in parallel etc.

Right we just invoke buildx binary for the inspect command integration test atm so we don't cover the `commands` package but don't think we need other kind of invocation of the integrations tests container like we do in BuildKit.

Reply:
> On BuildKit we needed this special case because we got to test the frontend specifically and also the gateway and I don't think we need to have this kind of logic in buildx.

I think our special case will come from wanting to test the commands specifically, and then also test the driver creation specifically. We'll probably want different types of environment for both (e.g. the driver creation tests will probably need to configure buildkit servers with different combinations of tls options, or spin up unique k8s clusters per test).

We might not do that though, so I think it's ok to simplify this for now, and we can always add it back in later.